dropai.services.yml

```diff
@@ -23,3 +23,12 @@ services:
   plugin.manager.dropai_tokenizer:
     class: Drupal\dropai\Plugin\DropaiTokenizerManager
     arguments: ['@container.namespaces', '@cache.discovery', '@module_handler']
+  # Define a service to build a Factory to the document Reader.
+  dropai.document_reader.factory:
+    class: Drupal\dropai\Service\DocReaderFactory
+    arguments: ['@config.factory']
+  # Define a service for media content access.
+  dropai.doc_reader.access_check.doc_content:
```

Comment:
This is working and is a common pattern in Drupal. But technically, this is not an issue of 'access'. Instead something like a plugin deriver would be useful for programmatically setting contextual links. This is worth learning about in the future but is not something we need to change right now.